### PR TITLE
chore: add dependabot

### PR DESCRIPTION
**Pull request by gssbzn**
## Description

This adds github dependabot, we can use this to keep dependencies in check

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

Example PR https://github.com/mongodb/mongocli/pull/345

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+  - package-ecosystem: gomod
+    directory: "/"
+    schedule:
+      interval: weekly
+      day: monday
```
Comment (gssbzn) [inline .github/dependabot.yml:7]: we can also auto assign people to the PRs but I think we should probably handle this with a group, @themantissa what do you think here?

Comment (themantissa) [reply .github/dependabot.yml:7]: A group makes it much less likely to get missed if someone is OOO - perhaps just internal MDBers on your team? I can be in the group as well if you'd like to ensure someone is keeping an eye out?

Comment (gssbzn) [reply .github/dependabot.yml:7]: yep, we can make this an internal devs group, I'm no admin here so just let me know the name of the group and I'll add it here

**Comment by themantissa** [comment]
@gssbzn created @mongodb/dependabot-team and added you.  I'm assuming this is what you mean by group?  If so I can add others as needed.

**Comment by gssbzn** [comment]
@themantissa FYI I did a run on this on mcli and you may need to invite the team to the repo

**Pull request merged**